benchmarks/helics/EchoLeafFederateMain.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright (c) 2017-2020,
+Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance for Sustainable Energy, LLC.  See
+the top-level NOTICE for additional details. All rights reserved.
+SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include "EchoLeafFederate.hpp"
+
+#include <chrono>
+#include <iostream>
+
+int main(int argc, char* argv[])
+{
+    helics::FederateInfo fi;
+    EchoLeaf fed;
+
+    int rc = fed.initialize(fi, argc, argv);
+    if (rc != 0) {
+        exit(rc);
+    }
+
+    // setup benchmark timing
+    std::chrono::time_point<std::chrono::steady_clock> start_time, end_time;
+    fed.setBeforeFinalizeCallback([&end_time]() { end_time = std::chrono::steady_clock::now(); });
+
+    // run the benchmark
+    fed.run([&start_time]() { start_time = std::chrono::steady_clock::now(); });
+
+    // print duration
+    auto elapsed =
+        std::chrono::duration_cast<std::chrono::nanoseconds>(end_time - start_time).count();
+    fed.addResult<decltype(elapsed)>("ELAPSED TIME (ns)", "real_time", elapsed);
+    fed.printResults();
```

Comment:
Is there going to be any difference in the output from the hub to leafs logs so we know which is which?

Reply:
There should be a difference -- do you think a default result entry called ("Federate Name") should be added to distinguish them, or should the launcher name the files with the federate type?

The file naming should probably be done either way, so it's mostly a question of whether the federate should print the name as part of the file contents.

Reply:
If the file name contains the differentiation is might not be required but I suppose it would be nice to be able to get all the information in single block, instead of having to parse the file name and the file.  If it was contained in the file the processor could separate the steps and just parsing the file contents would have all the information.  So I think unless there is some issue with generating it, I think that information should be in the file contents as well.

Reply:
I added a "benchmark federate type" output that shows the name of the federate app (EchoHub, EchoLeaf, etc).